staging/src/k8s.io/component-base/config/v1alpha1/defaults.go

```diff
@@ -44,7 +44,7 @@ func RecommendedDefaultLeaderElectionConfiguration(obj *LeaderElectionConfigurat
 		obj.RetryPeriod = metav1.Duration{Duration: 2 * time.Second}
 	}
 	if obj.ResourceLock == "" {
-		obj.ResourceLock = EndpointsResourceLock
+		obj.ResourceLock = EndpointsLeasesResourceLock
```

Comment:
Changing this default is problematic. It means any component running a 1.17 and 1.16 instance off the same config (like during a rolling upgrade) would lock on two different resources.

Reply:
Oh… just realized this is not switching to leases but to endpointsleases. Do we know everywhere this is used, and whether all consumers are accounting for the extra leases permissions required?

Reply:
It would be safer for the scheduler and controller-manager to change their defaults, rather than change this here

Reply:
Changed to scheduler and controller manager and left a TODO here to do that for everything.